pkg/messaging/broker/publisher.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package broker
+
+import (
+	"errors"
+
+	"github.com/mainflux/mainflux"
+	"github.com/mainflux/mainflux/pkg/messaging/nats"
+	"github.com/mainflux/mainflux/pkg/messaging/rabbitmq"
+)
+
+const (
+	envBrokerType = "MF_BROKER_TYPE"
+)
+
+var (
+	errEmptyBrokerType = errors.New("empty broker type")
+	defBrokerType      = "nats"
+)
+
+// NewPublisher This aggregates the NewPublisher function for all message brokers
+func NewPublisher(url string) (nats.Publisher, error) {
+	brokerSelection := mainflux.Env(envBrokerType, defBrokerType)
```

Comment:
It's probably better to receive broker type as a parameter and use `mainflux.Env` only in `main.go`. The reason is sneaky configuration bugs that can arise when loading config from env this way (encapsulated in the method itself).